### ISSUE TITLE
Support for Leveled Log Context Properties

### ISSUE THREAD

**Issue by pardahlman** [serilog/serilog#1093]
**Does this issue relate to a new *feature* or an existing *bug*?**
- [ ] Bug
- [x] New Feature

**What version of Serilog is affected?  Please list the related NuGet package.**

`Serilog`

**What is the target framework and operating system? See [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [x] netCore 2.0
- [ ] netCore 1.0
- [ ] 4.7
- [ ] 4.6.x
- [ ] 4.5.x

**Please describe the current behavior?**

When the output template contains `{Properties}`, all properties is added without any respect taken to log level.

**Please describe the expected behavior?**

I imagine that there are several good reasons why this is a bad idea, but here goes...! :bowtie: 

I recently created a new ASP.NET Core web application and configured Serilog to be my "application logger" (using `Serilog.AspNetCore`) with following configuration

```csharp
Log.Logger = new LoggerConfiguration()
  .MinimumLevel.Verbose()
  .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
  .Enrich.FromLogContext()
  .WriteTo.Console(outputTemplate: LogTemplates.Console)
  .CreateLogger();
```
My output template for the console contains `{Properties}` and it was bloated with ASP.NET internal properties (some of which I don't mind, but other felt a bit verbose).

This got me thinking that it would be a nice addition if properties appended to the log context could be "leveled", meaning that when I push a new enricher (like the `PropertyEnricher`) to the context, I could append a `LogEventLevel` and this level would be honored when rendering the log event to the different sinks.

I'm note sure how this would tie back to my original problem with the scope from the "application logger",as `BeginScope` does not imply any specific log level 🤷‍♂️.

Please tell me why this is not feasible 😉

**Comment by nblumhardt** [comment]
Thanks for the ticket.

I think we already have the basis for this in your PR https://github.com/serilog/serilog/pull/1034 (released with 2.6). But as you say, `BeginScope()` isn't levelled, so I don't know how this would work in the scenario you described.

**Comment by pardahlman** [comment]
But of course 🤦‍♂️ - I'll close this ticket, as I think there currently is no way to level the events from the application logger. Thanks!
